beets/util/__init__.py

```diff
@@ -398,7 +413,7 @@ def displayable_path(path, separator='; '):
         return path.decode('utf-8', 'ignore')
 
 
-def syspath(path, prefix=True):
+def syspath(path: Bytestring, prefix: bool = True) -> Bytestring:
```

Comment:
```suggestion
def syspath(path: bytes, prefix: bool = True) -> Union[bytes, str]:
```

Output type is platform-dependent, unfortunately. Maybe, there's a way to encode this is in the types?

Reply:
No, there isn't. That isn't how the typing works. The way to do that would be different functions that are typed differently. 

I defined `Bytestring` as equivalent to the union of strings and bytes. Any reason why you changed it? I think the return might be a string no matter what since the conversion happens but that has knock on effects with the rest of the code.

Also `path` can clearly be a string, not bytes, because there's an `isinstance` check in that function that sees whether it's in string or byte form.

Reply:
Oh, what happened here is that I confused the custom `Bytestring` with [`typing.ByteString`](https://docs.python.org/3/library/typing.html#typing.ByteString). Seems useful to have an alias here, but we may want to come up with a dfferent name.

> I think the return might be a string no matter what since the conversion happens but that has knock on effects with the rest of the code.

No, the conversion does not always happen, only on Windows.

> Also path can clearly be a string, not bytes, because there's an isinstance check in that function that sees whether it's in string or byte form.

You're right, that is indeed the case. From my understanding, correct usage of that function shouldn't pass `str`,  but it probably makes sense to keep that check as a safeguard for now.